zebra-utils/src/bin/zebra-checkpoints/args.rs

```diff
@@ -3,10 +3,40 @@
 //! For usage please refer to the program help: `zebra-checkpoints --help`
 
 use structopt::StructOpt;
+use thiserror::Error;
+
+use std::str::FromStr;
+
+/// The backend type the zebra-checkpoints utility will use to get data from.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum Backend {
+    Zebrad,
+    Zcashd,
+}
+
+impl FromStr for Backend {
```

Comment:
Optional/Nitpick: it would be nice to always use verbose=0 and deserialize the block to simplify the code.

Reply:
Downloading the entire block will significantly reduce performance, so I'd like to keep the faster mode available.

Reply:
I just checked the performance, and it seems pretty similar. So I am not concerned about the difference, particularly for a tool we only run every few weeks.

So I don't mind what we do here, but we're trying to minimise changes so we can get this PR merged and move on with other work.

Here are the performance figures, it turns out the new mode is actually the fastest overall. It can take almost twice as long to query `zcashd` with `--backend zebrad` on mainnet, otherwise the times aren't that different.

Mainnet:
```sh
$ time zebra-checkpoints-release --backend zcashd --last-checkpoint $(tail -1 zebra-consensus/src/checkpoint/main-checkpoints.txt | cut -d" " -f1) | tee /tmp/main-zcashd-zcashd.txt
1941792 0000000000e0989263bef30dd453c6813dee07bdd1b61d3c846fae98d2c8929b
...
1954447 000000000086aa0f9e3375ef2c9211b524ab562aba10024c2d4b1e856bf223cc
real    6m5.974s
$ time zebra-checkpoints-release --backend zebrad --last-checkpoint $(tail -1 zebra-consensus/src/checkpoint/main-checkpoints.txt | cut -d" " -f1) | tee /tmp/main-zebrad-zcashd.txt
...
real    9m27.998s
$ time zebra-checkpoints-release --backend zebrad --last-checkpoint $(tail -1 zebra-consensus/src/checkpoint/main-checkpoints.txt | cut -d" " -f1) -- --rpcport=28232 | tee /tmp/main-zebrad-zebrad.txt
...
real    5m0.404s
```

Testnet:
```sh
$ time zebra-checkpoints-release --backend zcashd --last-checkpoint $(tail -1 zebra-consensus/src/checkpoint/test-checkpoints.txt | cut -d" " -f1) --cli zcash-cli-testnet | tee /tmp/test-zcashd-zcashd.txt
2183200 00388d6db62115418ba764acc6db4aa77e1dc40219d2d447e11605c75824549e
...
2196400 006756f092109fc63ebb76a9b5376d4a6a7e8a73efac32a682490546f647bf90
real    2m32.585s
$ time zebra-checkpoints-release --backend zebrad --last-checkpoint $(tail -1 zebra-consensus/src/checkpoint/test-checkpoints.txt | cut -d" " -f1) --cli zcash-cli-testnet | tee /tmp/test-zebrad-zcashd.txt
...
real    2m39.516s
$ time zebra-checkpoints-release --backend zebrad --last-checkpoint $(tail -1 zebra-consensus/src/checkpoint/test-checkpoints.txt | cut -d" " -f1) -- --rpcport=38232 | tee /tmp/test-zebrad-zebrad.txt
...
real    1m31.989s
```

Reply:
For the full sync, I used `zebrad` concurrently in both modes, so its times might be slower.

Mainnet:
```sh
$ time zebra-checkpoints-release --backend zcashd | tee /tmp/main-zcashd-zcashd.txt
1954681 00000000015a8986b4664632ecbcd9421b03af430991077b029dc2f6eb39709c
real    496m35.946s
$ time zebra-checkpoints-release --backend zebrad | tee /tmp/main-zebrad-zcashd.txt
real    846m48.880s
$ time zebra-checkpoints-release --backend zebrad -- --rpcport=28232 | tee /tmp/main-zebrad-zebrad.txt
real    604m2.113s
```

Testnet:
```sh
$ time zebra-checkpoints-release --backend zcashd --cli zcash-cli-testnet | tee /tmp/test-zcashd-zcashd.txt
2196800 0006a24636968703c7407a4444d34e313568352dc1a1179006b8ca84b5e004bd
real    438m13.955s
$ time zebra-checkpoints-release --backend zebrad --cli zcash-cli-testnet | tee /tmp/test-zebrad-zcashd.txt
real    437m48.286s
$ time zebra-checkpoints-release --backend zebrad -- --rpcport=38232 | tee /tmp/test-zebrad-zebrad.txt
real    257m32.831s
```

Zebra is almost twice as fast on testnet, and slower by about 20% on mainnet.